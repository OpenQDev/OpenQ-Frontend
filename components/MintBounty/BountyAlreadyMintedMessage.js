import React from 'react';
import Link from 'next/link';
import useWeb3 from '../../hooks/useWeb3';

export default function BountyAlreadyMintedMessage({ bountyAddress, bountyId, claimed }) {

	const {safe} = useWeb3();
	return (
		<div className="flex flex-row items-center space-x-1">
			<div className="pt-4 text-white">Bounty is already {claimed? 'claimed': 'minted'}, view</div>
			<Link
				href={`/?address=${bountyAddress}}`}
				as={`/bounty/${bountyId}/${bountyAddress}`}
			>
				<a
					target={safe ? '_self' : '_blank'}
					rel="noreferrer"
					className="cursor-pointer text-link pt-4"
				>
					here.
				</a>
			</Link>
		</div>
	);
}