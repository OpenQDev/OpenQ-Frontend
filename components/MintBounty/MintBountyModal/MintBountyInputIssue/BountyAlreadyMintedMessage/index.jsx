import React from 'react';
import Link from 'next/link';

export default function BountyAlreadyMintedMessage({ bountyAddress, closed, id }) {
  return (
    <div className='flex flex-row items-center space-x-1'>
      <div className=''>
        Bounty is already {closed ? 'closed' : 'minted'}, view{' '}
        <Link
          href={`/?address=${id}/${bountyAddress}}`}
          as={`/contract/${id}/${bountyAddress}`}
          target={'_blank'}
          rel='noreferrer'
          className='cursor-pointer text-link-colour hover:underline'
        >
          <span data-testid='link'>here</span>
        </Link>
        .
      </div>
    </div>
  );
}
