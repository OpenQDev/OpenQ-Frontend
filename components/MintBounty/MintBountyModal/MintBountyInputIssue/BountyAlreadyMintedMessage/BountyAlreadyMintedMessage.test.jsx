/**
 * @vi-environment jsdom
 */
import React from 'react';
import BountyAlreadyMintedMessage from '.';
import renderer from 'react-test-renderer';
import Constants from '../../../../../test-utils/constant';

describe('BountyAlreadyMintedMessage', () => {
  const bounty = { ...Constants.bounty, id: Constants.bountyId };
  it('should match DOM Snapshot and msg "is already minted" ', () => {
    const tree = renderer.create(
      <BountyAlreadyMintedMessage id={bounty.id} closed={false} bountyAddress={bounty.bountyAddress} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should match DOM Snapshot and msg "closed" ', () => {
    const tree = renderer.create(
      <BountyAlreadyMintedMessage id={bounty.id} closed={true} bountyAddress={bounty.bountyAddress} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
