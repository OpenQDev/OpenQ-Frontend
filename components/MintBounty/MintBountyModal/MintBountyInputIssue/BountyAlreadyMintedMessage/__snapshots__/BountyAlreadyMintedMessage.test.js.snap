// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`BountyAlreadyMintedMessage should match DOM Snapshot and msg "closed"  1`] = `
<div
  className="flex flex-row items-center space-x-1"
>
  <div
    className=""
  >
    Bounty is already 
    closed
    , view
     
    <a
      className="cursor-pointer text-link-colour hover:underline"
      href="/contract/I_kwDOGWnnz85I9Agl/0x066efd87027a4cc4a21c57467f224ef98241d774"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      rel="noreferrer"
      target="_blank"
    >
      <span
        data-testid="link"
      >
        here
      </span>
    </a>
    .
  </div>
</div>
`;

exports[`BountyAlreadyMintedMessage should match DOM Snapshot and msg "is already minted"  1`] = `
<div
  className="flex flex-row items-center space-x-1"
>
  <div
    className=""
  >
    Bounty is already 
    minted
    , view
     
    <a
      className="cursor-pointer text-link-colour hover:underline"
      href="/contract/I_kwDOGWnnz85I9Agl/0x066efd87027a4cc4a21c57467f224ef98241d774"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      rel="noreferrer"
      target="_blank"
    >
      <span
        data-testid="link"
      >
        here
      </span>
    </a>
    .
  </div>
</div>
`;
